services/business/checkers/pod_checker.go

```diff
@@ -13,38 +13,32 @@ type PodChecker struct {
 const podsCheckerType = "pod"
 
 // Runs all checkers for Pod objects passed into the PodChecker
-func (checker PodChecker) Check() *models.IstioTypeValidations {
+func (checker PodChecker) Check() models.IstioValidations {
 	return checker.runIndividualChecks()
 }
 
 // Runs individual checks for each pod in pod checker.
-func (checker PodChecker) runIndividualChecks() *models.IstioTypeValidations {
-	typeValidations := models.IstioTypeValidations{}
-	if len(checker.Pods) == 0 {
-		return &typeValidations
-	}
-
-	nameValidations := models.IstioNameValidations{}
-	typeValidations[podsCheckerType] = &nameValidations
+func (checker PodChecker) runIndividualChecks() models.IstioValidations {
+	validations := models.IstioValidations{}
 
 	for _, pod := range checker.Pods {
 		validation := models.IstioValidation{
 			Name:       pod.ObjectMeta.Name,
 			ObjectType: podsCheckerType,
 			Valid:      true,
 		}
-		nameValidations[pod.ObjectMeta.Name] = &validation
 
-		checkers := checker.enabledCheckersFor(&pod)
-
-		for _, podChecker := range checkers {
+		for _, podChecker := range checker.enabledCheckersFor(&pod) {
 			checks, isValid := podChecker.Check()
 			validation.Checks = append(validation.Checks, checks...)
 			validation.Valid = validation.Valid && isValid
 		}
+
+		key := models.IstioValidationKey{ObjectType: podsCheckerType, Name: pod.ObjectMeta.Name}
```

Comment:
How is performed the key equality given we don't implement any kind of "equals" function? Memory address or field by field comparison?

Reply:
[Not explicitly answered but I understand from there https://blog.golang.org/go-maps-in-action that it's a recursive field-by-field comparison] ... which is ok then :)